### PR TITLE
Fix for failing Travis CI builds on new PRs

### PR DESCRIPTION
**Pull request by lmarniazman**
Fix for #46

### CODE REVIEW

**Comment by lmarniazman** [comment]
@hohwille I'm guessing the issue is just that I have to be given some permission on the sonar-devon4j-plugin project on sonarcloud. As the Travis build is initiated by me but I'm not authorized to start an analysis, it fails. I linked my GitHub with SonarCloud, maybe this will already resolve it.

**Comment by lmarniazman** [comment]
@hohwille I did some more digging and there are a few things that may have gone wrong:

1. Token was not encrypted properly. Did you use the --pro flag when encrypting to make sure it was using travis-ci.com and not travis-ci.org?
1. Token is not registered in Travis. I don't know if this applies to Travis, but I saw some threads where people had a similar issue in Jenkins and one of the suggestions was to register the token in their Jenkins instance.
1. I'm simply not authorized to run analysis on this project. This seems least likely to me as the project is listed as public on sonarcloud and the token we use in the travis.yml was generated by you

When I try to set the token in the command as another parameter (-Dsonar.login={TOKEN}) the build still fails but the error message is now
`Not authorized. Please check the properties sonar.login and sonar.password.`
sonar.password should not be mandatory when sonar.login is provided with a valid token, but even when I set sonar.password as another parameter with a blank value, it gives the same error, which brings us to the last possibility:

4. Token is not valid/expired. 

I guess we will just have to try each of these and observe how our Travis responds.

File: src/main/java/com/devonfw/ide/sonarqube/common/impl/DevonSonarDefinition.java [hunk 1]
```diff
@@ -56,7 +55,7 @@ private void addRule(NewRepository repository, Class<?> ruleClass) {
     rule.setType(RuleType.CODE_SMELL);
     rule.setStatus(RuleStatus.valueOf(ruleAnnotation.status().toUpperCase(Locale.US)));
     rule.setHtmlDescription(ruleAnnotation.description());
-    rule.setTemplate(AnnotationUtils.getAnnotation(ruleClass, RuleTemplate.class) != null);
+    rule.setTemplate(AnnotationUtils.getAnnotation(ruleClass, org.sonar.check.Rule.class) != null);
```
Comment (hohwille) [inline src/main/java/com/devonfw/ide/sonarqube/common/impl/DevonSonarDefinition.java]: any reason for not using an import? However, minimal nice to have...

Comment (lmarniazman) [reply src/main/java/com/devonfw/ide/sonarqube/common/impl/DevonSonarDefinition.java]: The problem was that it took the Rule class from the org.sonar.api.server.rule.RulesDefinition package and not from org.sonar.check.Rule as it should. Even when adding the import it is not 'used' and therefore automatically deleted beacuse of the IDEs save actions.

**Pull request merged**